contracts/simple_dex/lib.rs

```diff
@@ -513,4 +528,29 @@ mod simple_dex {
             SimpleDex::new()
         }
     }
+
+    #[cfg(test)]
+    mod test {
+        use proptest::prelude::*;
+
+        use super::*;
+
+        proptest! {
+            #[test]
+            fn rounding_benefits_dex(
+                balance_token_a in 1..1000u128,
+                balance_token_b in 1..1000u128,
+                pay_token_a in 1..1000u128
+            ) {
+                let get_token_b =
+                    SimpleDex::_out_given_in(pay_token_a, balance_token_a, balance_token_b, 0).unwrap();
```

Comment:
I think we shoud play with different `swap_fee_percentage` as that may affect the results.

Reply:
mo nor necessary - there will most likely never be a swap fee charged. Swap fee in most DEXes goes to LPs and there is only one here (us)

Reply:
as you wish., but I doubt it will ever be non-zero

Reply:
It may be the case for us but people may decide to use it as starting point for their own contracts where the fee will be non-zero and that will influence the "dust", etc.